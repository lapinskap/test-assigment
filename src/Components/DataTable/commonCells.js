import React from 'react';
import __ from '../../utils/Translations';
import priceFormatter from '../../utils/jsHelpers/priceFormatter';

// eslint-disable-next-line import/prefer-default-export
export const mapValueFromOptions = (options, key, optionValue = 'value', optionLabel = 'label') => {
  const renderEditable = (cellInfo) => {
    const value = cellInfo.row[key];
    if (!Array.isArray(value)) {
      const matchedOption = options.find((option) => option[optionValue] === value);
      const result = matchedOption ? matchedOption[optionLabel] : value;
      return (
        <div>
          {result}
        </div>
      );
    }

    return (
      <ul>
        {value.map((elValue) => {
          const matchedOption = options.find((option) => option[optionValue] === elValue);
          const result = matchedOption ? matchedOption[optionLabel] : `${__('Nieznany zas√≥b')}: ${elValue}`;
          return <li key={elValue}>{result}</li>;
        })}
      </ul>
    );
  };
  return renderEditable;
};

export const getDateCell = (key, withTime = false) => {
  const renderEditable = (cellInfo) => {
    const date = cellInfo.row[key];
    if (!date) {
      return null;
    }
    const dateObject = new Date(date);
    return (
      <div className="d-block w-100 text-center">
        {withTime ? dateObject.toLocaleString('pl-PL') : dateObject.toLocaleDateString('pl-PL')}
      </div>
    );
  };
  return renderEditable;
};

export const priceColumn = (cellInfo) => (
  <div className="d-block w-100 text-center">
    {priceFormatter(cellInfo.row[cellInfo.column.id])}
  </div>
);
