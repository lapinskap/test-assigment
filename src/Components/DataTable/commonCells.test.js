import React from 'react';
import renderer from 'react-test-renderer';
import { mapValueFromOptions } from './commonCells';
import '../../tests/setupTests';

describe('mapValueFromOptions', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with single value', () => {
    const options = [
      { value: 'option_1', label: 'Option 1' },
      { value: 'option_2', label: 'Option 2' },
      { value: 'option_3', label: 'Option 3' },
    ];
    const row = { test: 'option_1' };
    const Component = mapValueFromOptions(options, 'test');
    const tree = renderer
      .create(<Component row={row} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with array values', () => {
    const options = [
      { value: 'option_1', label: 'Option 1' },
      { value: 'option_2', label: 'Option 2' },
      { value: 'option_3', label: 'Option 3' },
    ];
    const row = { test: ['option_1', 'option_4'] };
    const Component = mapValueFromOptions(options, 'test');
    const tree = renderer
      .create(<Component row={row} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
