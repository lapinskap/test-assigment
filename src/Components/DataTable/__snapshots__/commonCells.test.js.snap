// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapValueFromOptions renders correctly with array values 1`] = `
<ul>
  <li>
    Option 1
  </li>
  <li>
    Nieznany zas√≥b: option_4
  </li>
</ul>
`;

exports[`mapValueFromOptions renders correctly with single value 1`] = `
<div>
  Option 1
</div>
`;
